spec/lib/logstash/java_integration_spec.rb

```diff
@@ -0,0 +1,84 @@
+# encoding: utf-8
+require "spec_helper"
+require "logstash/java_integration"
+
+describe "Java integration" do
+
+  context "type equivalence" do
+
+    it "ArrayList should report to be a Ruby Array" do
+      expect(Java::JavaUtil::ArrayList.new.is_a?(Array)).to be_true
+    end
+
+    it "ArrayList should be class equivalent to Ruby Array" do
+      expect do
+        case Java::JavaUtil::ArrayList.new
+        when Array
+          true
+        else
+          raise
+        end
+      end.not_to raise_error
+    end
+
+    it "LinkedHashMap should report to be a Ruby Hash" do
+      expect(Java::JavaUtil::LinkedHashMap.new.is_a?(Hash)).to be_true
+    end
+
+    it "LinkedHashMap should be class equivalent to Ruby Hash" do
+      expect do
+        case Java::JavaUtil::LinkedHashMap.new
+        when Hash
+          true
+        else
+          raise
+        end
+      end.not_to raise_error
+    end
+  end
```

Comment:
also, `kind_of` might be a correct test but these java integration patches were specifically made to solve the `case` and `is_a?` use-cases hence the specific test on `case` and `is_a?`.